Add local contribution guide (#69)
OpenTermsArchive/docs: content/contributing-terms.en.md

weight: 2
---

# Contributing new declarations

This is a step by step guide to help you add declarations to the [Contrib collection](https://github.com/OpenTermsArchive/contrib-declarations). This collection is dedicated to volunteer contribution of declarations to Open Terms Archive.

> Having understood briefly how a declaration is structured in JSON format, we need to look at concrete steps on how you can add these JSON files to the repository.

Reviewer: At this stage, the reader has not really “understood briefly how a declaration is structured in JSON format”.

Author: Made the change

## Prerequisites

In order to add declarations:

> 1. You need to have [Node.js](https://nodejs.org/en/) installed on your machine. If you don't have it, you can download it from the official website [here](https://nodejs.org/en/download/).
> 2. You need to have git installed on your machine. If you don't have it, you can download it from the official website [here](https://git-scm.com/downloads).

Reviewer: Prefer inline links 🙂 

```suggestion
1. You need to have [Node.js](https://nodejs.org/en/) installed on your machine. If you don't have it, you can [download it](https://nodejs.org/en/download/) from the official website.
2. You need to have Git installed on your machine. If you don't have it, you can [download it](https://git-scm.com/downloads) from the official website.
```

## Adding a declaration

To add a declaration, you need to follow these steps:

> 1. Clone the [contrib-declaration](https://github.com/OpenTermsArchive/contrib-declarations) repository to your local machine.

Reviewer: ```suggestion
1. Clone the [`contrib-declarations`](https://github.com/OpenTermsArchive/contrib-declarations) repository to your local machine.
```

> 2. Create a branch that describes your contribution e.g. `add-Open-Terms-Archive-ToS` or `add-firefox-privacy-policy`

Reviewer: Even though this is not mandatory, let's use the same format as the one used by the contribution tool 🙂 

```suggestion
2. Create a branch that describes your contribution e.g. `add_open_terms_archive_terms_of_service` or `add_firefox_privacy_policy`
```

Author: Yeah
It will help with the uniformity ✅

3. Run `npm install`. This will install all the dependencies including the Open Terms Archive engine which will allow you to test and validate your declaration.

> 4. Create a JSON file with, as filename, the service ID of the service you are adding the declaration for. This JSON file should be in the `declarations` folder of the repository. To learn more about selecting the right service ID, please read the [declaring a new service]({{< relref "reference/declaration#declaring-a-new-service" >}}) section.

Reviewer: Add links to the Declaration reference page for “Service ID”.

Author: Added inline link to that

  …
```

## Contributing new declarations

> This is a step by step guide to help you add declarations to the [contrib-declaration](https://github.com/OpenTermsArchive/contrib-declarations) repository. This repository is dedicated for volunteer contribution of declarations to Open Terms Archive.

Reviewer: While these are technically repositories, let's use the business term of “collection” as it describes more precisely what these repositories are intended for 🙂 

```suggestion
This is a step by step guide to help you add declarations to the [Contrib collection](https://github.com/OpenTermsArchive/contrib-declarations). This collection is dedicated to volunteer contribution of declarations to Open Terms Archive.
```

Having understood briefly how a declaration is structured in JSON format, we need to look at concrete steps on how you can add these JSON files to the repository. To add them, you need to:

1. Clone the [contrib-declaration](https://github.com/OpenTermsArchive/contrib-declarations) repository to your local machine.
2. Create a branch that describes your contribution e.g. `add-Open-Terms-Archive-ToS` or `add-firefox-privacy-policy`

> 3. Run `npm install`. This is to install all the dependencies including the Open Terms Archive engine which will allow you to test and validate your declaration to make sure it is ok.

Reviewer: ```suggestion
3. Run `npm install`. This will install all the dependencies including the Open Terms Archive engine which will allow you to test and validate your declaration.
```

Author: Thanks for the edits 😄

> 3. Run `npm install`. This is to install all the dependencies including the Open Terms Archive engine which will allow you to test and validate your declaration to make sure it is ok.

Reviewer: This assumes the existence of Node and NPM on the local machine 🙂 You probably need to add a step for that, or write it in a dedicated “prerequisites” section along with Git.

Author: Yeah 

Sure I will add that

> 4. Create a JSON file with the name of the service you are adding the declaration for. This JSON file should be in the `declarations` folder of the repository. To learn more about selecting the right service name, please read the [declaring a new service](#declaring-a-new-service) section of our docs.

Reviewer: The name of the JSON file should be the service ID, not the service name.
```suggestion
4. Create a JSON file with, as filename, the service ID of the service you are adding the declaration for. This JSON file should be in the `declarations` folder of the repository. To learn more about selecting the right service ID, please read the [declaring a new service](#declaring-a-new-service) section.
```

> 5. Visit the declaration URL and use [browser developer tools](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Tools_and_setup/What_are_browser_developer_tools) to inspect the page and find the right selectors for the significant section containing the terms you want to declare.

Reviewer: ```suggestion
5. Visit the declaration URL and use [browser developer tools](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Tools_and_setup/What_are_browser_developer_tools) to inspect the page and find the right selectors for the significant section containing the terms you want to declare and parts to remove.
```

> 6. After you've properly added your selectors and structured your JSON file, you need to test and validate your JSON file to make sure it is ok. To do this, you need to run `npx ota validate --services [service name]` from the root of the repository. This will run a validation on the declaration, highlighting any changes required.

Reviewer: ```suggestion
6. After you've properly chosen your selectors and structured your JSON file, you should test and validate your JSON file. To do this, you need to run `npx ota validate --services [service name]` from the root of the repository. This will run a validation on the declaration, highlighting any changes required.
```

> 7. If all tests are good, make a pull request to the main repository.  

Reviewer: ```suggestion
7. If all tests pass, open a pull request to the main repository.  
```

> You can read more about the [CLI](https://docs.opentermsarchive.org/#cli) to learn more about other tests and linting you can run on your declaration

Reviewer: ```suggestion
You can read more about the [CLI](https://docs.opentermsarchive.org/#cli) to learn more about other tests and linting you can run on your declaration.
```
